KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs

```diff
@@ -308,6 +309,10 @@ private string ProcessInlineContentItems(JToken linkedItems, Dictionary<string,
                             contentItem = GetContentItemModel(typeof(object), linkedItemsElementNode, linkedItemsNode, processedItems, currentlyResolvedRichStrings);
                             if (!processedItems.ContainsKey(codenameUsed))
                             {
+                                if (contentItem == null)
+                                {
+                                    contentItem = new UnknownContentItem(linkedItemsElementNode);
```

Comment:
Shouldn't we do this on the line 79 and react appropriately here?
```
if (t == null)
            {
                return null; // -> return new UnknownContentItem(item); ???
            }
```

Reply:
The public GetContentItemModel method would try to change the type from UnknownContentItem to T and that would cause an exception.

Reply:
hmm...good point.